packages/block-editor/src/components/rich-text/index.js

```diff
@@ -93,6 +93,10 @@ function getAllowedFormats( {
 
 getAllowedFormats.EMPTY_ARRAY = [];
 
+// The given text is considered a shortcode if
+// starts by the [ character and ends by the ] character.
+const isShortcode = ( text ) => new RegExp( '^\\[.*\\]$' ).test( text );
```

Comment:
Is this too naive a RegExp?

Reply:
You could grab [`regexp` from `@wordpress/shortcode`](https://github.com/WordPress/gutenberg/blob/68607318f23bfa371b7e1ec081dba88214d4ebd2/packages/shortcode/src/index.js#L155-L162) and [test it](https://github.com/WordPress/gutenberg/blob/68607318f23bfa371b7e1ec081dba88214d4ebd2/packages/blocks/src/api/raw-handling/shortcode-converter.js#L32), if you want the battle-tested stuff. :)

Reply:
nice! 8cf006b